Python/pystate.c

```diff
@@ -171,6 +171,65 @@ _PyRuntimeState_Fini(_PyRuntimeState *runtime)
 }
 
 #ifdef HAVE_FORK
+_Py_IDENTIFIER(stdout);
+_Py_IDENTIFIER(stderr);
+
+static int stdio_at_fork_reinit(_Py_Identifier *key)
+{
+
+    int ret = 0;
+
+    PyObject *stdio, *closed, *isatty, *buffer, *result;
+
+    stdio = _PySys_GetObjectId(key);
+
+    _Py_IDENTIFIER(closed);
+    if (_PyObject_LookupAttrId(stdio, &PyId_closed, &closed) < 0) {
+        PyErr_Clear();
+        goto end;
+    }
+
+    if (Py_IsTrue(closed)) {
+        goto end;
+    }
+
+    _Py_IDENTIFIER(isatty);
+    if (_PyObject_GetAttrId(stdio, &PyId_isatty) == NULL) {
+        PyErr_Clear();
+        goto end;
+    }
+
+    isatty = _PyObject_CallMethodIdNoArgs(stdio, &PyId_isatty);
+    if (Py_IsFalse(isatty)) {
+        goto end;
+    }
```

Comment:
I think this can be just one call to `_PyObject_CallMethodIdNoArgs` with a check for a NULL return value.